client/rest-high-level/src/main/java/org/elasticsearch/client/SecurityClient.java

```diff
@@ -98,6 +100,33 @@ public void putUserAsync(PutUserRequest request, RequestOptions options, ActionL
             PutUserResponse::fromXContent, listener, emptySet());
     }
 
+    /**
+     * Removes user from the native realm synchronously.
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-delete-user.html">
+     * the docs</a> for more.
+     * @param request the request with the user to delete
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the response from the delete user call
+     * @throws IOException in case there is a problem sending the request or parsing back the response
+     */
+    public DeleteUserResponse deleteUser(DeleteUserRequest request, RequestOptions options) throws IOException {
+        return restHighLevelClient.performRequestAndParseEntity(request, SecurityRequestConverters::deleteUser, options,
+            DeleteUserResponse::fromXContent, singleton(404));
```

Comment:
Hey I saw some updates on this PR and I just wanted to throw a reminder out that we are not going to do a singleton(404) here, because we want a delete that is not found to throw an exception. Also, last time we spoke you were going to change this to AcknowledgedResponse. <3

Reply:
It might also depend on which implementation of `AcknowledgedResponse` you are using, since we have two, yay duplication!!! You should have a look at `StartRollupJobResponse`, which is an example of changing the word from `acknowledged` to `started`. this should get you on the right track.

Reply:
`DeleteUserResponse` is a subclass of `AcknowledgedResponse`. I am using the same pattern.

Reply:
heh, duh... Sorry, ive been on vacation and full of turkey since last week.

Reply:
so the only thing here now is to remove `singleton(404)`, right?

Reply:
Apologies, I  misread the comment before. Since server side is not throwing exceptions, lets keep the `singleton(404)`, so we dont have to mod the parsing code. This is something we should definitely fix server side.